src/components/App.tsx

```diff
@@ -1,108 +1,33 @@
-import React, { useState, useEffect, useRef } from "react";
+import React from "react";
 
-import { useAccounts, useImageStorage } from "../hooks";
-import UploadPreviewModal from "./UploadPreviewModal";
-import UploadButton from "./UploadButton";
-import EmptyState from "./EmptyState";
-import ImageCard from "./ImageCard";
-import { ImageRecord } from "../types";
+import Login from "./Login";
+import ImageList from "./ImageList";
+import { useAccounts } from "../hooks";
 
 const App: React.FC = () => {
-	const [images, setImages] = useState<ImageRecord[]>([]);
-	const [selectedFile, setSelectedFile] = useState<File | null>(null);
-	const [preview, setPreview] = useState<string | null>(null);
-	const [isModalOpen, setIsModalOpen] = useState(false);
-	const { storeImage, fetchImages, isLoading } = useImageStorage();
-	const fileInputRef = useRef<HTMLInputElement>(null);
-
-	const accountHook = useAccounts();
-
-	useEffect(() => {
-		if (!isLoading) {
-			fetchImages().then(setImages);
-		}
-	}, [isLoading]);
-
-	useEffect(() => {
-		if (!selectedFile) {
-			setPreview(null);
-			return;
-		}
-
-		const objectUrl = URL.createObjectURL(selectedFile);
-		setPreview(objectUrl);
-		setIsModalOpen(true);
-
-		return () => {
-			URL.revokeObjectURL(objectUrl);
-		};
-	}, [selectedFile]);
-
-	const handleImageChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-		const file = e.target.files ? e.target.files[0] : null;
-		setSelectedFile(file);
-	};
-
-	// eslint-disable-next-line @typescript-eslint/no-unused-vars
-	const uploadImage = async (title?: string, _description?: string) => {
-		if (selectedFile) {
-			const encrypted = await accountHook.encrypt(title ?? "no title");
-			await storeImage(selectedFile, title, JSON.stringify(encrypted));
-			const updatedImages = await fetchImages();
-			setImages(updatedImages);
-			resetFormAndCloseModal();
-		}
-	};
-
-	const resetFormAndCloseModal = () => {
-		setSelectedFile(null);
-		setIsModalOpen(false);
-		if (fileInputRef.current) {
-			fileInputRef.current.value = "";
-		}
-	};
+	const { isLoggedIn, logout } = useAccounts();
 
 	return (
 		<div className="flex flex-col h-full">
 			{/* Top Navigation Bar */}
 			<div className="p-4 shadow-md fixed top-0 left-0 right-0 bg-white z-10">
 				<h1 className="text-center text-xl font-bold">Image App</h1>
+				{isLoggedIn ? (
```

Comment:
Instead of using a ternary, you can use `&&` so you don't need to write the empty `<></>` view

Reply:
Great call!